File: docs/breez-sdk/snippets/python/src/lnurl_withdraw.py
import logging
from breez_sdk_spark import (
    BreezClient,
    InputType,
    LnurlWithdrawRequest,
)


async def withdraw(client: BreezClient):
    # ANCHOR: lnurl-withdraw
    # Endpoint can also be of the form:
    # lnurlw://domain.com/lnurl-withdraw?key=val
    lnurl_withdraw_url = (
        "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekj"
        "mmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8"
        "qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk"
    )

    try:
        input_type = await client.parse(lnurl_withdraw_url)
        if isinstance(input_type, InputType.LNURL_WITHDRAW):
            # Amount to withdraw in sats between min/max withdrawable amounts
            amount_sats = 5_000
            withdraw_request = input_type[0]
            optional_completion_timeout_secs = 30

            request = LnurlWithdrawRequest(
                amount_sats=amount_sats,
                withdraw_request=withdraw_request,
                completion_timeout_secs=optional_completion_timeout_secs,
            )
            response = await client.lnurl().withdraw(request=request)


            payment = response.payment
            logging.debug(f"Payment: {payment}")
            return response
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: lnurl-withdraw

File: docs/breez-sdk/snippets/rust/src/lnurl_withdraw.rs
use breez_sdk_spark::*;
use anyhow::Result;
use log::info;

async fn withdraw(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: lnurl-withdraw
    // Endpoint can also be of the form:
    // lnurlw://domain.com/lnurl-withdraw?key=val
    let lnurl_withdraw_url = "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk";

    if let Ok(InputType::LnurlWithdraw(withdraw_request)) = sdk.parse(lnurl_withdraw_url).await {
        // Amount to withdraw in sats between min/max withdrawable amounts
        let amount_sats = 5_000;
        let optional_completion_timeout_secs = Some(30);

        let response = sdk
            .lnurl_withdraw(LnurlWithdrawRequest {
                amount_sats,
                withdraw_request,
                completion_timeout_secs: optional_completion_timeout_secs,
            })
            .await?;

        let payment = response.payment;
        info!("Payment: {payment:?}");
    }
    // ANCHOR_END: lnurl-withdraw

    Ok(())
}
